pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/event_handler.go

```diff
@@ -0,0 +1,162 @@
+}
+
+func enqueueService(namespace, serviceName string, q workqueue.RateLimitingInterface) {
+	if len(serviceName) == 0 || len(namespace) == 0 {
+		return
+	}
+
+	q.Add(reconcile.Request{
+		NamespacedName: types.NamespacedName{Namespace: namespace, Name: serviceName},
+	})
+}
+
+func handlePoolServiceUpdate(oldObject, newObject client.Object, q workqueue.RateLimitingInterface) {
+	oldPs := oldObject.(*v1alpha1.PoolService)
+	newPs := newObject.(*v1alpha1.PoolService)
+
+	oldServiceName := getServiceNameFromPoolService(oldPs)
+	newServiceName := getServiceNameFromPoolService(newPs)
+
+	if oldServiceName != newServiceName {
+		klog.Warningf("service name of %s/%s is changed from %s to %s", oldPs.Namespace, oldPs.Name, oldServiceName, newServiceName)
+		enqueueService(oldPs.Namespace, oldServiceName, q)
+		enqueueService(newPs.Namespace, newServiceName, q)
+		return
+	}
+	enqueueService(newPs.Namespace, newServiceName, q)
+
+	return
+}
+
+func NewNodePoolEventHandler(c client.Client) handler.EventHandler {
+	return handler.Funcs{
+		CreateFunc: func(createEvent event.CreateEvent, limitingInterface workqueue.RateLimitingInterface) {
+			allMultiRegionLbServicesEnqueue(c, limitingInterface)
+		},
+		UpdateFunc: func(updateEvent event.UpdateEvent, limitingInterface workqueue.RateLimitingInterface) {
+			handleNodePoolUpdate(c, updateEvent, limitingInterface)
+		},
+		DeleteFunc: func(deleteEvent event.DeleteEvent, limitingInterface workqueue.RateLimitingInterface) {
+			nodePoolRelatedServiceEnqueue(c, deleteEvent.Object, limitingInterface)
+		},
+		GenericFunc: func(genericEvent event.GenericEvent, limitingInterface workqueue.RateLimitingInterface) {
+			nodePoolRelatedServiceEnqueue(c, genericEvent.Object, limitingInterface)
+		},
+	}
+}
+
+func allMultiRegionLbServicesEnqueue(c client.Client, q workqueue.RateLimitingInterface) {
+	services := &v1.ServiceList{}
+	err := c.List(context.Background(), services)
+	if err != nil {
+		return
+	}
+
+	for _, svc := range services.Items {
+		if !isLoadBalancerSetService(&svc) {
+			continue
+		}
+		enqueueService(svc.Namespace, svc.Name, q)
+	}
+}
+
+func handleNodePoolUpdate(c client.Client, updateEvent event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	oldNp := updateEvent.ObjectOld.(*v1beta1.NodePool)
+	newNp := updateEvent.ObjectNew.(*v1beta1.NodePool)
+
+	if reflect.DeepEqual(oldNp.Labels, newNp.Labels) {
+		return
+	}
+
+	allMultiRegionLbServicesEnqueue(c, q)
+}
+
+func nodePoolRelatedServiceEnqueue(c client.Client, object client.Object, q workqueue.RateLimitingInterface) {
+	np := object.(*v1beta1.NodePool)
+	poolServiceList := &v1alpha1.PoolServiceList{}
+
+	listSelector := client.MatchingLabels{
+		network.LabelNodePoolName: np.Name,
+		labelManageBy:             names.LoadBalancerSetController}
+
+	if err := c.List(context.Background(), poolServiceList, listSelector); err != nil {
+		return
+	}
+
+	for _, item := range poolServiceList.Items {
+		if !hasServiceName(&item) {
```

Comment:
remove this if condition, because handlePoolServiceNormal has handled this condition.

Reply:
done